### PR TITLE
adding s3 extensions 

### PR DESCRIPTION
**Pull request by HaykManukyanAvetiky**
coming with new s3 adapter I have add s3 extensions as well

### CODE REVIEW

File: specification/extensions/s3.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "definitions": {
+        "S3DataSetExtension": {
+            "type": "object",
+            "properties": {
+                "Files": {
```
Comment (germanosin) [inline specification/extensions/s3.json:7]: please use lowercase fieldnames

File: specification/extensions/s3.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                "Bucket": {
+                    "type": "string"
+                },
+                "Key": {
+                    "type": "string"
+                },
+                "Region": {
+                    "type": "string"
+                },
+                "Avg. file size": {
```
Comment (germanosin) [inline specification/extensions/s3.json:19]: please use camelCase in case of multiple words

**Review by germanosin — CHANGES_REQUESTED** [review]

File: specification/extensions/s3.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    "type": "int"
+                },
+                "estimatedDatasetSize": {
+                    "type": "string"
+                },
+                "fileFormat": {
+                    "type": "string"
+                },
+                "partitionStyle": {
+                    "type": "string"
```
Comment (germanosin) [inline specification/extensions/s3.json:29]: could we use enum here?

**Review by germanosin — CHANGES_REQUESTED** [review]

File: specification/extensions/s3.json (hunk 3 above)
Comment (HaykManukyanAvetiky) [reply specification/extensions/s3.json]: done , pls check

**Pull request closed without merging**